### PR TITLE
Ab#7900/fix wording for no trial period

### PR DESCRIPTION
**Pull request by josejibin**
[ADO](https://dev.azure.com/S72/SHIFT72/_boards/board/t/Web%20team/Stories/?workitem=7900)

Zero is condsidered as plural in i18 lib. So existing logic won't work.

### CODE REVIEW

**Comment by additiverich** [comment]
@josejibin Is the implication that if the user is using (for example) Arabic, which treats "0" as a "zero" plurality, there won't be a translation for the `shopping_price_plan_note` if Count is 0?

File: site/ja_JP.all.json [hunk 1]
```diff
@@ -843,10 +843,12 @@
     "other": "価格"
   },
   "shopping_price_plan_note": {
-    "zero": "{{ .Interval }}ごとに課金されます。",
     "one": "24時間の無料トライアルが終了した後、.Intervalごとに{{ .Interval }}",
```
Comment (PeculiarGoat) [inline site/ja_JP.all.json:846]: Japanese only supports [other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L15).

File: site/lt_LT.all.json [hunk 2]
```diff
@@ -1082,10 +1082,12 @@
     "other": "Kaina"
   },
   "shopping_price_plan_note": {
-    "zero": "Įkraunama kas {{ .Interval }} .",
     "one": "{{ .Interval }} kas .Intervalas pasibaigus 24 valandų nemokamam bandomajam laikotarpiui.",
```
Comment (PeculiarGoat) [inline site/lt_LT.all.json:1085]: Lithuanian  supports [one, few, many, and other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L164).

File: site/ru_RU.all.json [hunk 3]
```diff
@@ -1023,10 +1023,12 @@
     "other": "Цена"
   },
   "shopping_price_plan_note": {
-    "zero": "{{ .Interval }} каждые .интервал.",
     "one": "{{ .Interval }} каждые после окончания 24-часового бесплатного пробного периода.",
```
Comment (PeculiarGoat) [inline site/ru_RU.all.json:1026]: Russian requires [one, few, many, and other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L176).

File: site/sr_SR.all.json [hunk 4]
```diff
@@ -1082,10 +1082,12 @@
     "other": "Цена"
   },
   "shopping_price_plan_note": {
-    "zero": "Наплаћује се сваки {{ .Interval }} .",
     "one": "Наплаћује се сваки {{ .Interval }} након завршетка бесплатног пробног периода од 24 сата.",
```
Comment (PeculiarGoat) [inline site/sr_SR.all.json:1085]: Serbian requires [one, few, and other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L108).

File: site/uk_UA.all.json [hunk 5]
```diff
@@ -1125,10 +1125,12 @@
     "other": "Ціна"
   },
   "shopping_price_plan_note": {
-    "zero": "Стягується через кожний {{ .Interval }} .",
     "one": "Стягується кожні {{ .Interval }} після закінчення 24-годинної безкоштовної пробної версії.",
```
Comment (PeculiarGoat) [inline site/uk_UA.all.json:1128]: Ukranian supports [one, few, many, other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L176).

File: site/zh_TW.all.json [hunk 6]
```diff
@@ -1169,10 +1169,12 @@
     "other": "價錢"
   },
   "shopping_price_plan_note": {
-    "zero": "每隔{{ .Interval }}充電。",
     "one": "24 小時免費試用期結束後， {{ .Interval }}收費一次。",
```
Comment (PeculiarGoat) [inline site/zh_TW.all.json:1172]: Chinese only supports [other](https://github.com/nicksnyder/go-i18n/blob/0dc1626d56435e9d605a29875701721c54bc9bbd/i18n/language/codegen/plurals.xml#L15).

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
Relish is able to fall back on `other` if the desired plurality isn't there, but Kibble doesn't... it will just return the key.

Even though `shopping_price_plan_note` is a translation for a Relish component, I think we should start beating these translation files into shape and enforce the plural rules before we move the data to the database.

**Comment by stajs** [comment]
@josejibin please request review from delivery team to auto-assign reviewers.

image

File: site/ar_LB.all.json [hunk 7]
```diff
@@ -1163,10 +1163,12 @@
     "other": "سعر"
   },
   "shopping_price_plan_note": {
-    "zero": "اتهم كل {{ .Interval }} .",
```
[the remark this answers is not in the document]
Comment (PeculiarGoat) [reply site/ar_LB.all.json:1166]: `shopping_price_plan_note` for Arabic is still missing `zero`.  Everything else is looking good though!

[the remark this answers is not in the document]
Comment (PeculiarGoat) [reply site/ar_LB.all.json:1166]: @josejibin I'll approve once you add a `zero` plurality to `shopping_price_plan_note`!

**Comment by additiverich** [comment]
(Just looping in some more delivery team members)

**Review by PeculiarGoat — APPROVED** [review]
Nice 👍

**Review by curtisadams28 — APPROVED** [review]

File: site/ar_LB.all.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (stajs) [reply site/ar_LB.all.json:1166]: @PeculiarGoat Perhaps we bake in some plurality validation in the API?

**Review by stajs — CHANGES_REQUESTED** [review]
Yikes, I know you've just merged and I've come in way late with the review.

**Pull request merged**